### ISSUE TITLE
update failed from 22.2.10~ynh1 to 24.0.4~ynh1

### ISSUE THREAD

**Issue by rocky-III** [YunoHost-Apps/nextcloud_ynh#504]
[YunoHost] 11.0.9.13
update failed from 22.2.10 ynh1 to 24.0.4 ynh1:

https://paste.yunohost.org/raw/decahokile

**Comment by utzer** [comment]
Question, why is version 23 skipped when Nexrcloud manual states that you should only update from the latest point release to a new major version, so from 23.0.x to 24.x is fine, but 22 to 24 is not allowed. 

Why is this skipped, anyone knows?

**Comment by ericgaspar** [comment]
We don't skip version... the upgrade script as a sequential upgrade for each major version to the next one (https://github.com/YunoHost-Apps/nextcloud_ynh/blob/0d102edec98fbb6c0518d850b01dfcb4b7079289/scripts/upgrade#L195-L200)
so we can upgrade from 22.x to 24.x

**Comment by Gredin67** [comment]
might be solved by #452

**Comment by Gredin67** [comment]
The error is related to php deps
```
2022-09-06 15:47:08,892: DEBUG - + apt-get --assume-yes --quiet -o=Acquire::Retries=3 -o=Dpkg::Use-Pty=0 --no-remove --option Dpkg::Options::=--force-confdef --option Dpkg::Options::=--force-confold install php8.0-fpm php8.0-bz2 php8.0-imap php8.0-gmp php8.0-intl php8.0-curl php8.0-apcu php8.0-redis php8.0-ldap php8.0-imagick php8.0-zip php8.0-igbinary php8.0-bcmath php8.0 --dry-run
2022-09-06 15:47:10,301: WARNING - The following packages have unmet dependencies:
2022-09-06 15:47:10,302: WARNING -  php8.0-igbinary : Breaks: php-igbinary (< 3.2.6+2.0.8-6+0~20220131.33+debian11~1.gbp1d540e~)
2022-09-06 15:47:10,304: WARNING -  php8.0-redis : Breaks: php-redis (< 5.3.7+4.3.0-1+0~20220330.42+debian11~1.gbp6fe8b7~)
2022-09-06 15:47:10,305: DEBUG - + ynh_die '--message=Unable to install dependencies'
2022-09-06 15:47:10,340: WARNING - Unable to install dependencies
2022-09-06 15:47:10,343: DEBUG - + ynh_exit_properly
2022-09-06 15:47:10,847: DEBUG - + ynh_clean_setup
2022-09-06 15:47:10,848: DEBUG - + ynh_restore_upgradebackup
2022-09-06 15:47:10,848: DEBUG - + ynh_print_err '--message=Upgrade failed.'
2022-09-06 15:47:10,870: WARNING - [Error] Upgrade failed.
```

**Comment by ericgaspar** [comment]
Should be fixed with #555
